File: isValid.cpp
class Solution {
public:
    bool isValid(string s) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
     	vector<char> stack;
     	for(int i = 0;i < s.size();i ++)
     	{
     		switch(s[i])
     		{
     			case '(':stack.push_back(s[i]);break;
     			case ')':
				{
					if(stack.size()==0 || stack.back()!='(')
					{
						return false;
					}
					else
					{
						stack.pop_back();
					}
					break;
				}
				case '{':stack.push_back(s[i]);break;
     			case '}':
				{
					if(stack.size()==0 || stack.back()!='{')
					{
						return false;
					}
					else
					{
						stack.pop_back();
					}
					break;
				}
				case '[':stack.push_back(s[i]);break;
     			case ']':
				{
					if(stack.size()==0 || stack.back()!='[')
					{
						return false;
					}
					else
					{
						stack.pop_back();
					}
					break;
				}
     		}
     	}
     	if(stack.size()==0)
     	{
     		return true;
     	}
     	return false;
    }
};

File: isValid.java
public class Solution {
    public boolean isValid(String s) {
        // Start typing your Java solution below
        // DO NOT write main() function
        Stack<Character> stack = new Stack<Character>();
        for(int i = 0;i < s.length();i++)
        {
        	char c = s.charAt(i);
        	switch(c)
        	{
        		case '(':
        		case '{':
        		case '[':
        			stack.push(c);
        			break;
        		case ')':
        		{
        			if(stack.empty() || stack.pop() != '(')
        				return false;
        			break;
        		}
        		case '}':
        		{
        			if(stack.empty() || stack.pop() != '{')
        				return false;
        			break;
        		}
        		case ']':
        		{
        			if(stack.empty() || stack.pop() != '[')
        				return false;
        			break;
        		}
        	}
        }
        if(stack.empty()) return true;
        return false;
    }
}